### ISSUE TITLE
is.sorted is slow for determining whether jval is sorted by key

### ISSUE THREAD

**Issue by MichaelChirico** [Rdatatable/data.table#4498]
From SO:

https://stackoverflow.com/questions/62019120/why-does-data-table-notation-for-column-retrieval-affect-speed/62028864#62028864

```
x <- as.data.table(as.character(rnorm(20000000,1,0.5)))
setkey(x,V1)

tic(); x[,.(V1)]; toc()
# 25.08 sec elapsed
```

(timing is even worse on my machine)

The bottleneck appears to be this line:

https://github.com/Rdatatable/data.table/blob/dd7609e83132e19c9be80d71c73e8a8f95e19b27/R/data.table.R#L1339

IINM we can tell the output is sorted because `V1` is the key and it appears as a `name` -- no need to compute the sort order all over again.

**Comment by jangorecki** [comment]
could you check if recomputing key in this case will be resolved by #4386?

**Comment by jangorecki** [comment]
I think we can move this issue to a next release. Is.sorted is now optimized so the issue is at least less painful now.
